coding/javascript.md

```diff
@@ -8,25 +8,158 @@ redirect_from:
 
 This document outlines the rules for writing JavaScript across our codebase.
 
-## Progressive Enhancement
+## General Principals
```

Comment:
Is it "principals" or "principles"? Or are those the same?

Reply:
Oh, good catch, thanks :)

Reply:
I read that three times and came to the conclusion that "principals" was indeed right, but it actually 100% should be "principles" hahaha

Reply:
![](https://media.giphy.com/media/l4pMattUYTTM7qpIk/source.gif)

Reply:
An easy way to remember.... "a (school) principal is your PAL"  😄